lib/search/duplicate_remover.rb

```diff
@@ -0,0 +1,36 @@
+module Search
+  class DuplicateRemover
+    attr_reader :index, :logger
+
+    def initialize(index:, logger: Logger.new($stdout))
+      @index = index
+      @logger = logger
+    end
+
+    def remove_duplicates(duplicates:)
+      duplicates.each do |duplicate|
+        if duplicate[:documents].all? { |doc| doc["updated_at"].nil? }
+          logger.info "None of the documents with content_item: #{duplicate[:content_id]} have an 'updated_at' field."
+          next
+        end
+
+        sorted_docs = duplicate[:documents].sort do |doc1, doc2|
+          next 0 if doc1["updated_at"] == doc2["updated_at"]
+          next 1 if doc1["updated_at"].nil?
+          next -1 if doc2["updated_at"].nil?
+
+          doc2["updated_at"] <=> doc1["updated_at"]
+        end
+
+        _, *tail = sorted_docs
+        tail.each do |doc|
+          Services.elasticsearch.delete_by_query(
+            index: index,
+            body: { query: { term: { link: doc["link"] } } },
+          )
+          logger.info "Deleted duplicate document: #{doc['link']}"
+        end
```

Comment:
I find this method really hard to follow - the sort function in particular is melting my brain! Could you refactor it into smaller methods and add tests for them?